### ISSUE TITLE
Question about training result

### ISSUE THREAD

**Issue by ewayuan** [UKPLab/sentence-transformers#1304]
I have the two trained sbert models, one with 4 epochs, and one with 10 epochs, and I used EmbeddingSimilarityEvaluator. 

All the eval metrics (s.t. cosine_pearson,cosine_spearman,euclidean_pearson, etc) for the model with 10 epochs all higher than the model with 4 epochs, but when I use it for the downstream task to calculate the similarity score on the test set, the performance of the model with 10 epochs is worth than the model with 4 epochs.  In this case, I don't know how to choose the best number of epochs for my task.

How could I choose the number of epochs based on the eval metrics (s.t. cosine_pearson,cosine_spearman,euclidean_pearson, etc)?

**Comment by nreimers** [comment]
I wrote about this here:
https://arxiv.org/abs/1803.09578

It happens quite often that some model is better on the dev set, but performs worse on the test set.

Often, the difference is not so large, so choosing any of the models is fine. I would not worry to much about tiny differences in the scores. They are mostly from noise.

**Comment by ewayuan** [comment]
> I wrote about this here: https://arxiv.org/abs/1803.09578
> 
> It happens quite often that some model is better on the dev set, but performs worse on the test set.
> 
> Often, the difference is not so large, so choosing any of the models is fine. I would not worry to much about tiny differences in the scores. They are mostly from noise.

In this case, should I continue to increase the number of epochs (20 epochs) to train and achieve a better performance?

**Comment by nreimers** [comment]
You can try, but at some point you overfit on the train set and further training is not helpful
